lib/pages/class_feedback/view/class_feedback_view.dart

```diff
@@ -0,0 +1,689 @@
+        .fetchPeople()
+        .then((teachers) => setState(() => classTeachers = teachers));
+
+    awaitFeedbackQuestions();
+  }
+
+  Future<Map<String, dynamic>> awaitFeedbackQuestions() async {
+    await Provider.of<FeedbackProvider>(context, listen: false)
+        .fetchQuestions()
+        .then((questions) => setState(() => feedbackQuestions = questions));
+
+    for (int i = 0; i <= feedbackQuestions.length; i++) {
+      responses.insert(i, '-1');
+      initialValues.insert(i, {
+        0: false,
+        1: false,
+        2: false,
+        3: false,
+        4: false,
+      });
+    }
+
+    return feedbackQuestions;
+  }
+
+  Widget autocompleteAssistant(BuildContext context) {
+    return Autocomplete<Person>(
+      key: const Key('AutocompleteAssistant'),
+      fieldViewBuilder: (BuildContext context,
+          TextEditingController textEditingController,
+          FocusNode focusNode,
+          VoidCallback onFieldSubmitted) {
+        textEditingController.text = selectedAssistant?.name;
+        return TextFormField(
+          controller: textEditingController,
+          decoration: InputDecoration(
+            labelText: S.of(context).labelAssistant,
+            prefixIcon: const Icon(FeatherIcons.user),
+          ),
+          focusNode: focusNode,
+          onFieldSubmitted: (String value) {
+            onFieldSubmitted();
+          },
+          validator: (_) {
+            if (textEditingController.text.isEmpty ?? true) {
+              return S.current.warningYouNeedToSelectAtLeastOne;
+            }
+            return null;
+          },
+        );
+      },
+      displayStringForOption: (Person person) => person.name,
+      optionsBuilder: (TextEditingValue textEditingValue) {
+        if (textEditingValue.text == '' || textEditingValue.text.isEmpty) {
+          return const Iterable<Person>.empty();
+        }
+        if (classTeachers.where((Person person) {
+          return person.name
+              .toLowerCase()
+              .contains(textEditingValue.text.toLowerCase());
+        }).isEmpty) {
+          final List<Person> inputTeachers = [];
+          final Person inputTeacher =
+              Person(name: textEditingValue.text.titleCase);
+          inputTeachers.add(inputTeacher);
+          return inputTeachers;
+        }
+
+        return classTeachers.where((Person person) {
+          return person.name
+              .toLowerCase()
+              .contains(textEditingValue.text.toLowerCase());
+        });
+      },
+      onSelected: (Person selection) {
+        formKey.currentState.validate();
+        setState(() {
+          selectedAssistant = selection;
+        });
+      },
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final personProvider = Provider.of<PersonProvider>(context);
+    final feedbackProvider = Provider.of<FeedbackProvider>(context);
+
+    final List<String> generalQuestionsRating =
```

Comment:
The entire code here is really complicated and hard to read, it must have been a pain to write too... Treating each question type separately like this, in one single place, makes the code very hard to maintain too - what if you want a new question type?

I suggest looking at pages/timetable/model/events so you can see how different event types are implemented. You have a main class (`UniEvent`) and then various events that implement it, each with their own special fields. When reading from firebase, we check for the existence of a special field (e.g. rrule exists in firebase => instantiate a `RecurringEvent`).

You should do something similar here, except you can just look for the "type" field to choose which class type to instantiate. That way you can have a single `Map<String, FeedbackQuestion> questions`, where `FeedbackQuestion` is the parent type for all question types. Then you define a widget that takes a question and just draws it based on the type and content (e.g. if it's a rating question, then a widget with text and underneath it the emojis). The code in this method should essentially read like:

```dart
List<Widget> children = [
  classWidget(),
  lecturerWidget(),
  assistantWidget(),
  acknowledgementWidget(),
]

forEach(final category in categories) {  // or section, whatever you wanna call it
  List<Widget> categoryChildren = [categoryHeader(category)];
  forEach(final question in questions.where((q) => q.category == category)) {
    categoryChildren.add(questionWidget(question));
  }
  children.add(Card(child: Column(children: categoryChildren)));
}

return
...
  body: ListView(
    ...
    children: children,
  )
...
```

Note: The question class can contain the answer too (in the format specific to the question type, e.g. rating questions have an integer value from 0 to 4, open ended questions have a string, maybe dropdowns have either an enum or an id representing a dropdown option? The question widget can then easily update the content of the class itself when the user interacts with it, so you can then use it outside of the widget too.